### PR TITLE
JESI-3014

### PR DESCRIPTION
**Pull request by AndreTheHunter**
JESI-3014
https://github.com/jesims/bindle/tree/JESI-3014
- [x] integrate with build-scripts
- [x] test in other projects
- [ ] review other projects first in case changes need to be made

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,28 @@
-# bindle
\ No newline at end of file
+# bindle
+
+Public build scripts.
```
Comment (BadAlgorithm) [inline README.md]: Should we allude to the fact we have a private build scripts?

perhaps something like
```suggestion
Build tool scripts.
```

**Review by BadAlgorithm — CHANGES_REQUESTED** [review]

**Review by BadAlgorithm — DISMISSED** [review]

**Review by BadAlgorithm — DISMISSED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#TODO use Clojure config
+version: 2.1
+aliases:
+  containers:
+    docker: &DOCKER
+      - image: jesiio/build-bus:latest
+        user: node
+        environment:
+          JVM_OPTS: -Xmx3200m
+          DEBUG: 1
```
Comment (axrs) [inline .circleci/config.yml]: `JVM_OPTS` and `DEBUG` shouldn't be required.

**Review by BadAlgorithm — DISMISSED** [review]

File: README.md [hunk 3]
```diff
@@ -1 +1,28 @@
...
+
+```sh
+git submodule add git@github.com:jesims/bindle.git
+git submodule init
+```
+
+Then in your own project bash file (e.g. `jesi.sh`):
+
+```sh
+source bindle/project.sh
```
Comment (axrs) [inline README.md]: Add the `|| exit 1`?

**Review by axrs — CHANGES_REQUESTED** [review]
Smol

**Review by axrs — DISMISSED** [review]

**Review by BadAlgorithm — DISMISSED** [review]

File: project.sh [hunk 4]
```diff
@@ -319,6 +317,18 @@ is-lein () {
...
+is-java () {
+  file-exists 'pom.xml'
+}
+
+is-dry () {
+	file-exists 'package-dry.json'
+}
+
+is-npm () {
+	file-exists 'package.json'
```
Comment (BadAlgorithm) [inline project.sh:334]: Indentation here is inconsistent

**Review by BadAlgorithm — CHANGES_REQUESTED** [review]

File: project.sh [hunk 5]
```diff
@@ -319,6 +317,18 @@ is-lein () {
 	file-exists 'project.clj'
 }
 
+is-java () {
+	file-exists 'pom.xml'
```
Comment (axrs) [inline project.sh:326]: check pom.xml exists and project.clj does not?

File: project.sh [hunk 6]
```diff
@@ -356,32 +366,56 @@ lint-bash () {
...
 		require-committed .
 	fi
 }
 
+-outdated () {
+	if is-lein;then
+		lein-dev ancient check :all 2>/dev/null &&
+		lein-dev pom
+	fi
+	if is-java;then
```
Comment (axrs) [inline project.sh:418]: won't this do a lein, generate pom, then run the java as-well?

**Review by axrs — COMMENTED** [review]
💬

File: project.sh [hunk 7]
```diff
@@ -356,32 +366,56 @@ lint-bash () {
...
+	lein-lint || true
 	if is-ci;then
 		require-committed .
 	fi
 }
 
+-outdated () {
+	if is-lein;then
+		lein-dev ancient check :all 2>/dev/null &&
+		lein-dev pom
```
Comment (AndreTheHunter) [inline project.sh]: FIXME combine with `lein do`

File: project.sh (hunk 6 above)
Comment (AndreTheHunter) [reply project.sh:418]: Yup. `lein ancient` doesn't pick up all the java maven dependencies, so it's good to use maven to check also
Do lein then java, like `-deps` does

File: project.sh (hunk 5 above)
Comment (AndreTheHunter) [reply project.sh:326]: Nah, if a project needs that it can go `is-lein && ! is-java`

**Review by axrs — DISMISSED** [review]

**Review by BadAlgorithm — DISMISSED** [review]

**Review by BadAlgorithm — APPROVED** [review]

**Review by axrs — APPROVED** [review]

**Pull request merged**